<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>MuSiC: Multi-subject Single-cell Deconvolution</title>

<script src="MuSiC_files/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="MuSiC_files/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="MuSiC_files/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="MuSiC_files/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="MuSiC_files/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="MuSiC_files/navigation-1.1/tabsets.js"></script>
<link href="MuSiC_files/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="MuSiC_files/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>



<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div id="header">



<h1 class="title toc-ignore">MuSiC: Multi-subject Single-cell Deconvolution</h1>
<h4 class="author">Xuran Wang</h4>
<address class="author_afil">
Graduate Group of AMCS, University of Pennsylvania, Philadelphia, USA<br>
</div>


<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>This vignette provides a walk through tutorial on how to use <code>MuSiC</code> to estimate cell type proportions from bulk sequencing data based on multi-subject single cell data by reproducing the analysis in MuSiC paper, now is published on <a href="https://doi.org/10.1038/s41467-018-08023-x"><em>Nature Communications</em></a>.</p>
<div id="installation" class="section level2">
<h2>Installation</h2>
<p>To run the entire deconvolution tutorial, users need to install the <a href="https://www.github.com/xuranw/MuSiC"><code>MuSiC</code></a> package.</p>
<pre class="r"><code># install devtools if necessary
install.packages(&#39;devtools&#39;)

# install the MuSiC package
devtools::install_github(&#39;xuranw/MuSiC&#39;)

# load
library(MuSiC)</code></pre>
</div>
<div id="data" class="section level2">
<h2>Data</h2>
<p><code>MuSiC</code> uses two types of input data:</p>
<ul>
<li><p>Bulk expression obtained from RNA sequencing, which is a mixture expression of various cell types. These are the data we want to deconvolve.</p></li>
<li><p>Multi-subject single cell expression obtained from single-cell RNA sequencing (scRNA-seq). The cell types of scRNA-seq are pre-determined. These serve as reference for estimating cell type proportions of bulk data.</p></li>
</ul>
<p><code>MuSiC</code> requires raw read counts for both bulk and single-cell expression. The discussion of the usage of RPKM and TPM can be found in the <strong>Discussion</strong> section of our <a href="https://doi.org/10.1038/s41467-018-08023-x">paper</a>.</p>
</div>
</div>
<div id="music-deconvolution" class="section level1">
<h1>MuSiC Deconvolution</h1>
<p>MuSiC utilizes cell-type specific gene expression from single-cell RNA sequencing (RNA-seq) data to characterize cell type compositions from bulk RNA-seq data in complex tissues. By appropriate weighting of genes showing cross-subject and cross-cell consistency, MuSiC enables the transfer of cell type-specific gene expression information from one dataset to another.</p>
<p>Solid tissues often contain closely related cell types which leads to collinearity. To deal with collinearity, MuSiC employs a tree-guided procedure that recursively zooms in on closely related cell types. Briefly, we first group similar cell types into the same cluster and estimate cluster proportions, then recursively repeat this procedure within each cluster.</p>
<p><img src="images/FigureMethod.jpg" /></p>
</div>
<div id="sample-analysis" class="section level1">
<h1>Sample Analysis</h1>
<div id="data-summary" class="section level2">
<h2>Data Summary</h2>
<p>This vignette reproduces the <strong>human pancreatic islet</strong> and the <strong>mouse kidney</strong> analysis, which require single cell and bulk RNA-seq datasets from following sources:</p>
<table>
<colgroup>
<col width="13%" />
<col width="24%" />
<col width="13%" />
<col width="20%" />
<col width="17%" />
<col width="12%" />
</colgroup>
<thead>
<tr class="header">
<th>Datasets</th>
<th><span class="citation">Segerstolpe et al. (2016)</span></th>
<th><span class="citation">Xin et al. (2016)</span></th>
<th><span class="citation">Fadista et al. (2014)</span></th>
<th><span class="citation">Park et al. (2018)</span></th>
<th><span class="citation">Beckerman et al. (2017)</span></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Session Num.</td>
<td><a href="https://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-5061/">E-MTAB-5061</a></td>
<td><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE81608">GSE81608</a></td>
<td><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE50244">GSE50244</a></td>
<td><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE107585">GSE107585</a></td>
<td><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE81492">GSE81492</a></td>
</tr>
<tr class="even">
<td>Data type</td>
<td>scRNA-seq</td>
<td>scRNA-seq</td>
<td>bulk RNA-seq</td>
<td>scRNA-seq</td>
<td>bulk RNA-seq</td>
</tr>
<tr class="odd">
<td>Protocol</td>
<td>Smart-seq2</td>
<td>Illumina HiSeq 2500</td>
<td>Illumina HiSeq 2000</td>
<td>10x</td>
<td>Illumina HiSeq 2500</td>
</tr>
<tr class="even">
<td>Tissue</td>
<td><strong>human pancreas</strong></td>
<td><strong>human pancreas</strong></td>
<td><strong>human pancreas</strong></td>
<td><strong>mouse kidney</strong></td>
<td><strong>mouse kidney</strong></td>
</tr>
<tr class="odd">
<td># of genes</td>
<td>25,453</td>
<td>39,849</td>
<td>56,638</td>
<td>16,273</td>
<td>19,033</td>
</tr>
<tr class="even">
<td># of cells</td>
<td>2,209</td>
<td>1,492</td>
<td>NA</td>
<td>43,745</td>
<td>NA</td>
</tr>
<tr class="odd">
<td># of cell types</td>
<td>14 + 1 NA</td>
<td>4</td>
<td>NA</td>
<td>14 + 2 novel</td>
<td>NA</td>
</tr>
<tr class="even">
<td># of subjects</td>
<td>10 (6 healthy + 4 T2D)</td>
<td>18 (12 healthy + 6 T2D)</td>
<td>89</td>
<td>7</td>
<td>10 (6 control + 4 APOL1)</td>
</tr>
</tbody>
</table>
<p>Bioconductor base package provides <code>ExpressionSet</code> class, which is a convenient data structure to hold expression data along with sample/feature annotation. Here we use two <code>ExpressionSet</code> objects to handle the bulk and single cell data respectively. The details of constructing <code>ExpressionSet</code> can be found on <a href="https://www.rdocumentation.org/packages/Biobase/versions/2.32.0/topics/ExpressionSet">this page</a>. Please see the answer of this <a href="https://github.com/xuranw/MuSiC/issues/2">Issue</a> for a simple guidance.</p>
<p>Datasets described in the table above are all in the form of <code>ExpressionSet</code> and available at the <a href="pages/data.html">data download page</a>.</p>
</div>
<div id="estimation-of-cell-type-proportions" class="section level2">
<h2>Estimation of cell type proportions</h2>
<p>Instead of selecting marker genes, MuSiC gives weights to each gene. The weighting scheme is based on cross-subject variation: up-weigh genes with low variation and down-weigh genes with high variation. Here we demonstrate step by step with the human pancreas datasets.</p>
<div id="data-preparations" class="section level3">
<h3>Data Preparations</h3>
<div id="bulk-data-of-human-pancreas" class="section level4">
<h4>Bulk data of human pancreas</h4>
<p>The dataset from <span class="citation">Fadista et al. (2014)</span> contains raw read counts data from bulk RNA-seq of human pancreatic islets to study glucose metabolism in healthy and hyper-hypoglycemic conditions. For the purpose of this vignette, the dataset is pre-processed and made available on the <a href="pages/data.html">data download page</a>. In addition to read counts, this dataset also contains HbA1c levels, BMI, gender and age information for each subject.</p>
<pre class="r"><code>GSE50244.bulk.eset = readRDS(&#39;https://xuranw.github.io/MuSiC/data/GSE50244bulkeset.rds&#39;)
GSE50244.bulk.eset
#ExpressionSet (storageMode: lockedEnvironment)
#assayData: 32581 features, 89 samples 
#  element names: exprs 
#protocolData: none
#phenoData
#  sampleNames: Sub1 Sub2 ... Sub89 (89 total)
#  varLabels: sampleID SubjectName ... tissue (7 total)
#  varMetadata: labelDescription
#featureData: none
#experimentData: use &#39;experimentData(object)&#39;
#Annotation:  </code></pre>
</div>
<div id="single-cell-data-of-human-pancreas" class="section level4">
<h4>Single cell data of human pancreas</h4>
<p>The single cell data are from <span class="citation">Segerstolpe et al. (2016)</span>, which constrains read counts for 25453 genes across 2209 cells. Here we only include the 1097 cells from 6 healthy subjects. The read counts are available on the <a href="pages/data.html">data download page</a>, in the form of an <code>ExpressionSet</code>.</p>
<pre class="r"><code># Download EMTAB single cell dataset from Github
EMTAB.eset = readRDS(&#39;https://xuranw.github.io/MuSiC/data/EMTABesethealthy.rds&#39;)
EMTAB.eset
#ExpressionSet (storageMode: lockedEnvironment)
#assayData: 25453 features, 1097 samples 
#  element names: exprs 
#protocolData: none
#phenoData
#  sampleNames: AZ_A10 AZ_A11 ... HP1509101_P9 (1097 total)
#  varLabels: sampleID SubjectName cellTypeID cellType
#  varMetadata: labelDescription
#featureData: none
#experimentData: use &#39;experimentData(object)&#39;
#Annotation:  </code></pre>
<p>Another single cell data is from <span class="citation">Xin et al. (2016)</span>, which have 39849 genes and 1492 cells. The read counts are available on the <a href="pages/data.html">data download page</a>, in the form of an <code>ExpressionSet</code>.</p>
<pre class="r"><code># Download Xin et al. single cell dataset from Github
XinT2D.eset = readRDS(&#39;https://xuranw.github.io/MuSiC/data/XinT2Deset.rds&#39;)
XinT2D.eset
#ExpressionSet (storageMode: lockedEnvironment)
#assayData: 39849 features, 1492 samples 
#  element names: exprs 
#protocolData: none
#phenoData
#  sampleNames: Sample_1 Sample_2 ... Sample_1492 (1492 total)
#  varLabels: sampleID SubjectName ... Disease (5 total)
#  varMetadata: labelDescription
#featureData: none
#experimentData: use &#39;experimentData(object)&#39;
#Annotation:  </code></pre>
</div>
</div>
<div id="bulk-tissue-cell-type-estimation" class="section level3">
<h3>Bulk Tissue Cell Type Estimation</h3>
<p>The deconvolution of 89 subjects from <span class="citation">Fadista et al. (2014)</span> are preformed with bulk data <code>GSE50244.bulk.eset</code> and single cell reference <code>EMTAB.eset</code>. We constrained our estimation on 6 major cell types: alpha, beta, delta, gamma, acinar and ductal, which make up over 90% of the whole islet.</p>
<p>The cell type proportions are estimated by the function <a href="../reference/music_prop.html"><code>music_prop</code></a>. The essential inputs are</p>
<ul>
<li><code>bulk.eset</code>: ExpressionSet of bulk data;</li>
<li><code>sc.eset</code>: ExpressionSet of annotated single cell data;</li>
<li><code>markers</code>: vector or list of gene names, default as <code>NULL</code>. If <code>NULL</code>, then use all genes that overlapping in bulk and single-cell dataset;</li>
<li><code>clusters</code>: character, must be one of the column names of the phenoData from single cell data, used as clusters;</li>
<li><code>samples</code>: character, must be one of the column names of the phenoData from single cell data, used as samples;</li>
<li><code>select.ct</code> vector of cell types included for deconvolution, default as <code>NULL</code>. If <code>NULL</code>, then use all cell types that provided by single-cell dataset.</li>
<li><code>verbose</code> logical that toggles log messages.</li>
</ul>
<p>The function <a href="../reference/music_prop.html"><code>music_prop</code></a> provides output as a list with elements:</p>
<ul>
<li><code>Est.prop.weighted</code>: data.frame of MuSiC estimated proportions, subjects by cell types;</li>
<li><code>Est.prop.allgene</code>: data.frame of NNLS estimated proportions, subjects by cell types;</li>
<li><code>Weight.gene</code>: matrix, MuSiC estimated weight for each gene, genes by subjects;</li>
<li><code>r.squared.full</code>: vector of R squared from MuSiC estimated proportions for each subject;</li>
<li><code>Var.prop</code>: matrix of variance of MuSiC estimates.</li>
</ul>
<p>The estimated proportions are normalized to sum to 1 across included cell types. Here we use <code>GSE50244.bulk.eset</code> as the <code>bulk.eset</code> input and <code>EMTAB.eset</code> as <code>sc.eset</code> input. The <code>clusters</code> is specified as <code>cellType</code> while <code>samples</code> is <code>sampleID</code>. As stated before, we only included 6 major cell types as <code>select.ct</code>.</p>
<pre class="r"><code># Estimate cell type proportions
Est.prop.GSE50244 = music_prop(bulk.eset = GSE50244.bulk.eset, sc.eset = EMTAB.eset, clusters = &#39;cellType&#39;,
                               samples = &#39;sampleID&#39;, select.ct = c(&#39;alpha&#39;, &#39;beta&#39;, &#39;delta&#39;, &#39;gamma&#39;,
                                                                   &#39;acinar&#39;, &#39;ductal&#39;), verbose = F)
names(Est.prop.GSE50244)
#[1] &quot;Est.prop.weighted&quot; &quot;Est.prop.allgene&quot;  &quot;Weight.gene&quot;       &quot;r.squared.full&quot;    &quot;Var.prop&quot;  </code></pre>
<p>Here we use <a href="../reference/Prop_comp_multi.html"><code>Jitter_Est</code></a> to show the difference between different estimation methods.</p>
<pre class="r"><code># Jitter plot of estimated cell type proportions
jitter.fig = Jitter_Est(list(data.matrix(Est.prop.GSE50244$Est.prop.weighted),
                             data.matrix(Est.prop.GSE50244$Est.prop.allgene)),
                        method.name = c(&#39;MuSiC&#39;, &#39;NNLS&#39;), title = &#39;Jitter plot of Est Proportions&#39;)

# A more sophisticated jitter plot is provided as below. We seperated the T2D subjects and normal 
#subjects by their HbA1c levels.

m.prop.GSE50244 = rbind(melt(GSE50244.EMTAB.prop$Est.prop.weighted), 
                        melt(GSE50244.EMTAB.prop$Est.prop.allgene))

colnames(m.prop.GSE50244) = c(&#39;Sub&#39;, &#39;CellType&#39;, &#39;Prop&#39;)
m.prop.GSE50244$CellType = factor(m.prop.GSE50244$CellType, levels = c(&#39;alpha&#39;, &#39;beta&#39;, &#39;delta&#39;, &#39;gamma&#39;, &#39;acinar&#39;, &#39;ductal&#39;))
m.prop.GSE50244$Method = factor(rep(c(&#39;MuSiC&#39;, &#39;NNLS&#39;), each = 89*6), levels = c(&#39;MuSiC&#39;, &#39;NNLS&#39;))
m.prop.GSE50244$HbA1c = rep(GSE50244.bulk.eset$hba1c, 2*6)
m.prop.GSE50244 = m.prop.GSE50244[!is.na(m.prop.GSE50244$HbA1c), ]
m.prop.GSE50244$Disease = factor(c(&#39;Normal&#39;, &#39;T2D&#39;)[(m.prop.GSE50244$HbA1c &gt; 6.5)+1], levels = c(&#39;Normal&#39;, &#39;T2D&#39;))
m.prop.GSE50244$D = (m.prop.GSE50244$Disease == &#39;T2D&#39;)/5
m.prop.GSE50244 = rbind(subset(m.prop.GSE50244, Disease == &#39;Normal&#39;), subset(m.prop.GSE50244, Disease != &#39;Normal&#39;))

jitter.new = ggplot(m.prop.GSE50244, aes(Method, Prop)) + 
  geom_point(aes(fill = Method, color = Disease, stroke = D, shape = Disease), 
             size = 2, alpha = 0.7, position = position_jitter(width=0.25, height=0)) +
  facet_wrap(~ CellType, scales = &#39;free&#39;) + scale_colour_manual( values = c(&#39;white&#39;, &quot;gray20&quot;)) +
  scale_shape_manual(values = c(21, 24))+ theme_minimal()

plot_grid(jitter.fig, jitter.new, labels = &#39;auto&#39;)</code></pre>
<p><img src="images/GSE50244jitter.jpg" /></p>
<p>It is well known that the beta cell proportions is related to T2D disease status. In the progress of T2D, the number of beta cells decreases. One of the most important test for T2D is HbA1c (hemoglobin A1c) test. When HbA1c level is greater than 6.5%, the patient is diagnosed as T2D. Let’s look at the beta cell proportions with HbA1c level.</p>
<pre class="r"><code># Create dataframe for beta cell proportions and HbA1c levels
m.prop.ana = data.frame(pData(GSE50244.bulk.eset)[rep(1:89, 2), c(&#39;age&#39;, &#39;bmi&#39;, &#39;hba1c&#39;, &#39;gender&#39;)],
                        ct.prop = c(Est.prop.GSE50244$Est.prop.weighted[, 2], 
                                    Est.prop.GSE50244$Est.prop.allgene[, 2]), 
                        Method = factor(rep(c(&#39;MuSiC&#39;, &#39;NNLS&#39;), each = 89), 
                        levels = c(&#39;MuSiC&#39;, &#39;NNLS&#39;)))
colnames(m.prop.ana)[1:4] = c(&#39;Age&#39;, &#39;BMI&#39;, &#39;HbA1c&#39;, &#39;Gender&#39;)
m.prop.ana = subset(m.prop.ana, !is.na(HbA1c))
m.prop.ana$Disease = factor( c(&#39;Normal&#39;, &#39;T2D&#39;)[(m.prop.ana$HbA1c &gt; 6.5) + 1], c(&#39;Normal&#39;, &#39;T2D&#39;) )
m.prop.ana$D = (m.prop.ana$Disease == &#39;T2D&#39;)/5

ggplot(m.prop.ana, aes(HbA1c, ct.prop)) + 
  geom_smooth(method = &#39;lm&#39;,  se = FALSE, col = &#39;black&#39;, lwd = 0.25) +
  geom_point(aes(fill = Method, color = Disease, stroke = D, shape = Disease), size = 2, alpha = 0.7) +  facet_wrap(~ Method) + 
  ggtitle(&#39;HbA1c vs Beta Cell Type Proportion&#39;) + theme_minimal() + 
  scale_colour_manual( values = c(&#39;white&#39;, &quot;gray20&quot;)) + 
  scale_shape_manual(values = c(21, 24))</code></pre>
<p><img src="images/GSE50244beta.jpg" /></p>
<p>The numerical evaluation can be obtained by linear regression. There is a significant negative correlation between HbA1c levels and beta cell proportions, after adjusted Age, BMI and Gender.</p>
<pre class="r"><code>lm.beta.MuSiC = lm(ct.prop ~ HbA1c + Age + BMI + Gender, data = subset(m.prop.ana, Method == &#39;MuSiC&#39;))
lm.beta.NNLS = lm(ct.prop ~ HbA1c + Age + BMI + Gender, data = subset(m.prop.ana, Method == &#39;NNLS&#39;))
summary(lm.beta.MuSiC)
#
#Call:
#lm(formula = ct.prop ~ HbA1c + Age + BMI + Gender, data = subset(m.prop.ana, 
#    Method == &quot;MuSiC&quot;))
#
#Residuals:
#     Min       1Q   Median       3Q      Max 
#-0.27768 -0.13186 -0.01096  0.10661  0.35790 
#
#Coefficients:
#              Estimate Std. Error t value Pr(&gt;|t|)    
#(Intercept)   0.877022   0.190276   4.609 1.71e-05 ***
#HbA1c        -0.061396   0.025403  -2.417   0.0182 *  
#Age           0.002639   0.001772   1.489   0.1409    
#BMI          -0.013620   0.007276  -1.872   0.0653 .  
#GenderFemale -0.079874   0.039274  -2.034   0.0457 *  
#---
#Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
#
#Residual standard error: 0.167 on 72 degrees of freedom
#Multiple R-squared:  0.2439,   Adjusted R-squared:  0.2019 
#F-statistic: 5.806 on 4 and 72 DF,  p-value: 0.0004166

summary(lm.beta.NNLS)
#
#Call:
#lm(formula = ct.prop ~ HbA1c + Age + BMI + Gender, data = subset(m.prop.ana, 
#    Method == &quot;NNLS&quot;))
#
#Residuals:
#     Min       1Q   Median       3Q      Max 
#-0.04671 -0.02918 -0.01795  0.01394  0.19362 
#
#Coefficients:
#               Estimate Std. Error t value Pr(&gt;|t|)  
#(Intercept)   0.0950960  0.0546717   1.739   0.0862 .
#HbA1c        -0.0093214  0.0072991  -1.277   0.2057  
#Age           0.0005268  0.0005093   1.035   0.3044  
#BMI          -0.0015116  0.0020906  -0.723   0.4720  
#GenderFemale -0.0037650  0.0112844  -0.334   0.7396  
#---
#Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
#
#Residual standard error: 0.04799 on 72 degrees of freedom
#Multiple R-squared:  0.0574,   Adjusted R-squared:  0.005028 
#F-statistic: 1.096 on 4 and 72 DF,  p-value: 0.3651</code></pre>
</div>
</div>
<div id="estimation-of-cell-type-proportions-with-pre-grouping-of-cell-types" class="section level2">
<h2>Estimation of cell type proportions with pre-grouping of cell types</h2>
<p>Solid tissues often contain closely related cell types, and correlation of gene expression between these cell types leads to collinearity, making it difficult to resolve their relative proportions in bulk data. To deal with collinearity, MuSiC employs a tree-guided procedure that recursively zooms in on closely related cell types. Briefly, we first group similar cell types into the same cluster and estimate cluster proportions, then recursively repeat this procedure within each cluster. At each recursion stage, we only use genes that have low within-cluster variance, a.k.a. the cross-cell consistent genes. This is critical as the mean expression estimates of genes with high variance are affected by the pervasive bias in cell capture of scRNA-seq experiments, and thus cannot serve as reliable reference.</p>
<p>We demonstrate this procedure by reproducing the analysis of mouse kidney in MuSiC <a href="https://doi.org/10.1038/s41467-018-08023-x">paper</a>.</p>
<div id="data-preparation" class="section level3">
<h3>Data Preparation</h3>
<div id="bulk-data" class="section level4">
<h4>Bulk data</h4>
<p>The dataset <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE81492">GEO entry (GSE81492)</a> <span class="citation">(see Beckerman et al. 2017)</span> contains raw RNA-seq and sample annotation data. For the purpose of this vignette, we will use the read counts data <code>Mousebulkeset.rds</code> from the <a href="pages/data.html">data download page</a>.</p>
<pre class="r"><code># Download Mouse bulk dataset from Github
Mouse.bulk.eset = readRDS(&#39;https://xuranw.github.io/MuSiC/data/Mousebulkeset.rds&#39;)
Mouse.bulk.eset

#ExpressionSet (storageMode: lockedEnvironment)
#assayData: 19033 features, 10 samples 
#  element names: exprs 
#protocolData: none
#phenoData
#  sampleNames: control.NA.27 control.NA.30 ... APOL1.GNA78M (10 total)
#  varLabels: sampleID SubjectName Control
#  varMetadata: labelDescription
#featureData: none
#experimentData: use &#39;experimentData(object)&#39;
#Annotation:  </code></pre>
</div>
<div id="single-cell-data" class="section level4">
<h4>Single cell data</h4>
<p>The single cell data are from <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE107585">GEO entry (GSE107585)</a> <span class="citation">(see Park et al. 2018)</span>, which constrains read counts for 16273 genes across 43745 cells. Due to the space limitation of Github, only a subset of the read counts <code>Mousesubeset.rds</code> are available on the <a href="page/data.html">data download page</a>, in the form of an <code>ExpressionSet</code>. This subset contains 16273 genes across 10000 cells.</p>
<pre class="r"><code># Download EMTAB single cell dataset from Github
Mousesub.eset = readRDS(&#39;https://xuranw.github.io/MuSiC/data/Mousesubeset.rds&#39;)
Mousesub.eset

#ExpressionSet (storageMode: lockedEnvironment)
#assayData: 16273 features, 10000 samples 
#  element names: exprs 
#protocolData: none
#phenoData
#  sampleNames: TGGTTCCGTCGGCTCA-2 CGAGCCAAGCGTCAAG-4 ... GAGCAGAGTCAACATC-1 (10000 total)
#  varLabels: sampleID SubjectName cellTypeID cellType
#  varMetadata: labelDescription
#featureData: none
#experimentData: use &#39;experimentData(object)&#39;
#Annotation:  

levels(Mousesub.eset$cellType)
# [1] &quot;Endo&quot;     &quot;Podo&quot;     &quot;PT&quot;       &quot;LOH&quot;      &quot;DCT&quot;      &quot;CD-PC&quot;    &quot;CD-IC&quot;    &quot;CD-Trans&quot; &quot;Novel1&quot;  
#[10] &quot;Fib&quot;      &quot;Macro&quot;    &quot;Neutro&quot;   &quot;B lymph&quot;  &quot;T lymph&quot;  &quot;NK&quot;       &quot;Novel2&quot;  </code></pre>
<p>Notice that the single cell dataset has 16 cell types, including 2 novel cell types and a transition cell type (CD-Trans). We exclude those 3 cell types in our analysis.</p>
</div>
</div>
<div id="estimation-of-cell-type-proportions-1" class="section level3">
<h3>Estimation of cell type proportions</h3>
<div id="clustering-single-cell-data" class="section level4">
<h4>Clustering single cell data</h4>
<p>In <a href="MuSiC.html#estimation-of-cell-type-proportions">previous</a> MuSiC estimation procedure, the first step is to produce design matrix, cross-subject mean of relative abundance, cross-subject variance of relative abundance and average library size from single cell reference. These are taken care of by the function <a href="../reference/music_basis.html"><code>music_basis</code></a>. The essential inputs of <a href="../reference/music_basis.html"><code>music_basis</code></a> are:</p>
<ul>
<li><code>x</code>: ExpressionSet of single cell dataset;</li>
<li><code>non.zero</code>: logical, default as <code>TRUE</code>. If true, remove all genes with zero expression;</li>
<li><code>markers</code>: vector or list of gene names. Default as <code>NULL</code>. If <code>NULL</code>, then use all genes provided by <code>x</code>;</li>
<li><code>clusters</code>: character, must be one of the column names of the phenoData from single cell data, used as clusters;</li>
<li><code>samples</code>: character, must be one of the column names of the phenoData from single cell data, used as samples;</li>
<li><code>select.ct</code> vector of cell types included for deconvolution, default as <code>NULL</code>. If <code>NULL</code>, then use all cell types that provided by single-cell dataset.</li>
<li><code>verbose</code> logical that toggles log messages.</li>
</ul>
<p>The outputs of <a href="../reference/music_basis.html"><code>music_basis</code></a> is a list of elements:</p>
<ul>
<li><code>Disgn.mtx</code>: gene by cell type matrix of design matrix;</li>
<li><code>S</code>: subject by cell type matrix of library size;</li>
<li><code>M.S</code>: vector of average library size for each cell type;</li>
<li><code>M.theta</code>: gene by cell type matrix of average relative abundance;</li>
<li><code>Sigma</code>: gene by cell type matrix of cross-subject variation.</li>
</ul>
<p>We next use the <code>hclust</code> function to get a tree0based clustering of the cell types using the cross-subject mean matrix and the design matrix.</p>
<pre class="r"><code># Produce the first step information
Mousesub.basis = music_basis(Mousesub.eset, clusters = &#39;cellType&#39;, samples = &#39;sampleID&#39;, 
                             select.ct = c(&#39;Endo&#39;, &#39;Podo&#39;, &#39;PT&#39;, &#39;LOH&#39;, &#39;DCT&#39;, &#39;CD-PC&#39;, &#39;CD-IC&#39;, &#39;Fib&#39;,
                                           &#39;Macro&#39;, &#39;Neutro&#39;,&#39;B lymph&#39;, &#39;T lymph&#39;, &#39;NK&#39;))

# Plot the dendrogram of design matrix and cross-subject mean of realtive abundance
par(mfrow = c(1, 2))
d &lt;- dist(t(log(Mousesub.basis$Disgn.mtx + 1e-6)), method = &quot;euclidean&quot;)
# Hierarchical clustering using Complete Linkage
hc1 &lt;- hclust(d, method = &quot;complete&quot; )
# Plot the obtained dendrogram
plot(hc1, cex = 0.6, hang = -1, main = &#39;Cluster log(Design Matrix)&#39;)
d &lt;- dist(t(log(Mousesub.basis$M.theta + 1e-8)), method = &quot;euclidean&quot;)
# Hierarchical clustering using Complete Linkage
# hc2 &lt;- hclust(d, method = &quot;complete&quot; )
hc2 &lt;- hclust(d, method = &quot;complete&quot;)
# Plot the obtained dendrogram
plot(hc2, cex = 0.6, hang = -1, main = &#39;Cluster log(Mean of RA)&#39;)</code></pre>
<p><img src="images/Cluster.jpg" /></p>
<p>The immune cells are clustered together and the kidney specific cells are clustered together. Notice that DCT and PT are within the same high-level grouping. The cut-off is user determined. Here we cut 13 cell types into 4 groups:</p>
<ul>
<li>Group 1: Neutro</li>
<li>Group 2: Podo</li>
<li>Group 3: Endo, CD-PC, CD-IC, LOH, DCT, PT</li>
<li>Group 4: Fib, Macro, NK, B lymph, T lymph</li>
</ul>
<p>The tree-guided recursive estimation for mouse kidney analysis includes 2 steps:</p>
<ul>
<li>Step 1. Estimate proportions of each high level cluster; <span class="math inline">\((p_1,p_2,p_3,p_4)\)</span></li>
<li>Step 2. Estimate cell type proportions within each cluster; <span class="math inline">\((p_{31},p_{32},.,p_{36},p_{41},.,p_{45})\)</span></li>
</ul>
</div>
<div id="bulk-tissue-cell-type-estimation-with-pre-grouping-of-cell-types" class="section level4">
<h4>Bulk Tissue Cell Type Estimation with Pre-grouping of Cell Types</h4>
<p>We manually specify the cluster and annotated single cell data with cluster information.</p>
<pre class="r"><code>clusters.type = list(C1 = &#39;Neutro&#39;, C2 = &#39;Podo&#39;, C3 = c(&#39;Endo&#39;, &#39;CD-PC&#39;, &#39;LOH&#39;, &#39;CD-IC&#39;, &#39;DCT&#39;, &#39;PT&#39;), C4 = c(&#39;Macro&#39;, &#39;Fib&#39;, &#39;B lymph&#39;, &#39;NK&#39;, &#39;T lymph&#39;))

cl.type = as.character(Mousesub.eset$cellType)

for(cl in 1:length(clusters.type)){
  cl.type[cl.type %in% clusters.type[[cl]]] = names(clusters.type)[cl]
}
pData(Mousesub.eset)$clusterType = factor(cl.type, levels = c(names(clusters.type), &#39;CD-Trans&#39;, &#39;Novel1&#39;, &#39;Novel2&#39;))

# 13 selected cell types
s.mouse = unlist(clusters.type)
s.mouse
#       C1        C2       C31       C32       C33       C34       C35       C36       C41       C42 
# &quot;Neutro&quot;    &quot;Podo&quot;    &quot;Endo&quot;   &quot;CD-PC&quot;     &quot;LOH&quot;   &quot;CD-IC&quot;     &quot;DCT&quot;      &quot;PT&quot;   &quot;Macro&quot;     &quot;Fib&quot; 
#      C43       C44       C45 
#&quot;B lymph&quot;      &quot;NK&quot; &quot;T lymph&quot; </code></pre>
<p>We then select genes that are differentially expressed within cluster <code>C3</code> (Epithelial cells) and <code>C4</code> (Immune cells), available on <a href="pages/data.html">data download page</a>. Function <a href="../reference/music_prop.cluster.html"><code>music_prop.cluster</code></a> is used for estimation with pre-clustering of cell types. The essential inputs are the same as <code>music_prop</code> except two unique inputs: <code>groups</code> and <code>group.markers</code>. <code>groups</code> passes the column name of higher-cluster in phenoData. The intra-cluster differentially expressed genes are passed by <code>group.marker</code>.</p>
<pre class="r"><code>load(&#39;https://xuranw.github.io/MuSiC/data/IEmarkers.RData&#39;)
# This RData file provides two vectors of gene names Epith.marker and Immune.marker

# We now construct the list of group marker
IEmarkers = list(NULL, NULL, Epith.marker, Immune.marker)
names(IEmarkers) = c(&#39;C1&#39;, &#39;C2&#39;, &#39;C3&#39;, &#39;C4&#39;)
# The name of group markers should be the same as the cluster names

Est.mouse.bulk = music_prop.cluster(bulk.eset = Mouse.bulk.eset, sc.eset = Mousesub.eset, group.markers = IEmarkers, clusters = &#39;cellType&#39;, groups = &#39;clusterType&#39;, samples = &#39;sampleID&#39;, clusters.type = clusters.type)</code></pre>
<p>Due to the limited space of Github, we can only demo <code>music_prop.cluster</code> with a subset of mouse kidney single cell dataset. Therefore, the results might be different from the one presented in the paper due to incomplete reference single cell dataset.</p>
</div>
</div>
</div>
<div id="benchmark-evaluation" class="section level2">
<h2>Benchmark Evaluation</h2>
<p>Benchmark dataset is constructed by summing up single cell data from <code>XinT2D.eset</code>. The artificial bulk data is constructed through function <a href="../reference/bulk_construct.html"><code>bulk_construct</code></a>. The inputs are single cell dataset, cluster name (<code>clusters</code>), sample name (<code>samples</code>) and selected cell type (<code>select.ct</code>). <a href="../reference/bulk_construct.html"><code>bulk_construct</code></a> returns a ExpressionSet of artificial bulk dataset <code>Bulk.counts</code> and a matrix of real cell type counts <code>num.real</code>.</p>
<pre class="r"><code># Construct artificial bulk dataset. Use all 4 cell types: alpha, beta, gamma, delta
XinT2D.construct.full = bulk_construct(XinT2D.eset, clusters = &#39;cellType&#39;, samples = &#39;SubjectName&#39;)

names(XinT2D.construct.full)
# [1] &quot;Bulk.counts&quot; &quot;num.real&quot; 

XinT2D.construct.full$Bulk.counts
#ExpressionSet (storageMode: lockedEnvironment)
#assayData: 39849 features, 18 samples 
#element names: exprs 
#protocolData: none
#phenoData
#sampleNames: Non T2D 1 Non T2D 2 ... T2D 6 (18 total)
#varLabels: SubjectName sampleID Disease
#varMetadata: labelDescription
#featureData: none
#experimentData: use &#39;experimentData(object)&#39;
#Annotation:  

head(XinT2D.construct.full$num.real)
#          alpha beta delta gamma
#Non T2D 1    53   13     5     3
#Non T2D 2     4   13     2     5
#Non T2D 3    27   10     3     2
#Non T2D 4    14   10     0     3
#Non T2D 5    23   22     5     2
#Non T2D 6    36    7     4     1

# calculate cell type proportions
XinT2D.construct.full$prop.real = relative.ab(XinT2D.construct.full$num.real, by.col = FALSE)
head(XinT2D.construct.full$prop.real)
#              alpha      beta      delta      gamma
#Non T2D 1 0.7162162 0.1756757 0.06756757 0.04054054
#Non T2D 2 0.1666667 0.5416667 0.08333333 0.20833333
#Non T2D 3 0.6428571 0.2380952 0.07142857 0.04761905
#Non T2D 4 0.5185185 0.3703704 0.00000000 0.11111111
#Non T2D 5 0.4423077 0.4230769 0.09615385 0.03846154
#Non T2D 6 0.7500000 0.1458333 0.08333333 0.02083333</code></pre>
<p>The estimation of cell type proportions:</p>
<pre class="r"><code># Estimate cell type proportions of artificial bulk data
Est.prop.Xin = music_prop(bulk.eset = XinT2D.construct.full$Bulk.counts, sc.eset = EMTAB.eset,
                          clusters = &#39;cellType&#39;, samples = &#39;sampleID&#39;, 
                          select.ct = c(&#39;alpha&#39;, &#39;beta&#39;, &#39;delta&#39;, &#39;gamma&#39;))
#Creating Relative Abudance Matrix...
#Creating Variance Matrix...
#Creating Library Size Matrix...
#Used 17884 common genes...
#Used 4 cell types in deconvolution...
#Non T2D 1 has common genes 15115 ...
#Non T2D 2 has common genes 12826 ...
#Non T2D 3 has common genes 13553 ...
#Non T2D 4 has common genes 13491 ...
#Non T2D 5 has common genes 14585 ...
#Non T2D 6 has common genes 13984 ...
#Non T2D 7 has common genes 13065 ...
#Non T2D 8 has common genes 13756 ...
#Non T2D 9 has common genes 13813 ...
#Non T2D 10 has common genes 14566 ...
#Non T2D 11 has common genes 14197 ...
#Non T2D 12 has common genes 15507 ...
#T2D 1 has common genes 14936 ...
#T2D 2 has common genes 15084 ...
#T2D 3 has common genes 14175 ...
#T2D 4 has common genes 16163 ...
#T2D 5 has common genes 15866 ...
#T2D 6 has common genes 15673 ...</code></pre>
<p>The numeric evaluation is conducted by <a href="../reference/Eval_multi.html"><code>Eval_multi</code></a>, which compares the real and estimated cell type proportions by</p>
<ol style="list-style-type: decimal">
<li>root-mean-squared deviation (RMSD);</li>
<li>mean absolute difference (mAD);</li>
<li>Pearson’s correlation (R).</li>
</ol>
<p>The visualization of cell type proportions are provided by <a href="../reference/Prop_comp_multi.html"><code>Prop_comp_multi</code></a>, <a href="../reference/Abs_diff_multi.html"><code>Abs_diff_multi</code></a> and <a href="../reference/Scatter_multi.html"><code>Scatter_multi</code></a>.</p>
<pre class="r"><code># Estimation evaluation

Eval_multi(prop.real = data.matrix(XinT2D.construct.full$prop.real), 
           prop.est = list(data.matrix(Est.prop.Xin$Est.prop.weighted), 
                           data.matrix(Est.prop.Xin$Est.prop.allgene)), 
           method.name = c(&#39;MuSiC&#39;, &#39;NNLS&#39;))

#           RMSD     mAD      R
#MuSiC   0.09881 0.06357 0.9378
#NNLS    0.17161 0.11749 0.8159

library(cowplot)
prop.comp.fig = Prop_comp_multi(prop.real = data.matrix(XinT2D.construct.full$prop.real),
                                prop.est = list(data.matrix(Est.prop.Xin$Est.prop.weighted),
                                                data.matrix(Est.prop.Xin$Est.prop.allgene)),
                                method.name = c(&#39;MuSiC&#39;, &#39;NNLS&#39;), 
                                title = &#39;Heatmap of Real and Est. Prop&#39; )

abs.diff.fig = Abs_diff_multi(prop.real = data.matrix(XinT2D.construct.full$prop.real),
                              prop.est = list(data.matrix(Est.prop.Xin$Est.prop.weighted),
                                              data.matrix(Est.prop.Xin$Est.prop.allgene)),
                              method.name = c(&#39;MuSiC&#39;, &#39;NNLS&#39;), 
                              title = &#39;Abs.Diff between Real and Est. Prop&#39; )

plot_grid(prop.comp.fig, abs.diff.fig, labels = &quot;auto&quot;, rel_widths = c(4,3))</code></pre>
<p><img src="images/benchmark.jpg" /></p>
<p>In our <a href="https://doi.org/10.1038/s41467-018-08023-x">paper</a>, we also compared our method with existing methods: <a href="https://cibersort.stanford.edu/">CIBERSORT</a> <span class="citation">(see Newman et al. 2015)</span> and <a href="https://github.com/shenorrLab/bseqsc">bseq-sc</a> <span class="citation">(see Baron et al. 2016)</span>.</p>
</div>
</div>
<div id="reference" class="section level1 unnumbered">
<h1>Reference</h1>
<div id="refs" class="references">
<div id="ref-baron2016single">
<p>Baron, Maayan, Adrian Veres, Samuel L Wolock, Aubrey L Faust, Renaud Gaujoux, Amedeo Vetere, Jennifer Hyoje Ryu, et al. 2016. “A Single-Cell Transcriptomic Map of the Human and Mouse Pancreas Reveals Inter-and Intra-Cell Population Structure.” <em>Cell Systems</em> 3 (4): 346–60.</p>
</div>
<div id="ref-beckerman2017transgenic">
<p>Beckerman, Pazit, Jing Bi-Karchin, Ae Seo Deok Park, Chengxiang Qiu, Patrick D Dummer, Irfana Soomro, Carine M Boustany-Kari, et al. 2017. “Transgenic Expression of Human Apol1 Risk Variants in Podocytes Induces Kidney Disease in Mice.” <em>Nature Medicine</em> 23 (4): 429.</p>
</div>
<div id="ref-fadista2014global">
<p>Fadista, João, Petter Vikman, Emilia Ottosson Laakso, Inês Guerra Mollet, Jonathan Lou Esguerra, Jalal Taneera, Petter Storm, et al. 2014. “Global Genomic and Transcriptomic Analysis of Human Pancreatic Islets Reveals Novel Genes Influencing Glucose Metabolism.” <em>Proceedings of the National Academy of Sciences</em> 111 (38): 13924–9.</p>
</div>
<div id="ref-newman2015robust">
<p>Newman, Aaron M, Chih Long Liu, Michael R Green, Andrew J Gentles, Weiguo Feng, Yue Xu, Chuong D Hoang, Maximilian Diehn, and Ash A Alizadeh. 2015. “Robust Enumeration of Cell Subsets from Tissue Expression Profiles.” <em>Nature Methods</em> 12 (5): 453.</p>
</div>
<div id="ref-park2018single">
<p>Park, Jihwan, Rojesh Shrestha, Chengxiang Qiu, Ayano Kondo, Shizheng Huang, Max Werth, Mingyao Li, Jonathan Barasch, and Katalin Suszták. 2018. “Single-Cell Transcriptomics of the Mouse Kidney Reveals Potential Cellular Targets of Kidney Disease.” <em>Science</em>, eaar2131.</p>
</div>
<div id="ref-segerstolpe2016single">
<p>Segerstolpe, Åsa, Athanasia Palasantza, Pernilla Eliasson, Eva-Marie Andersson, Anne-Christine Andréasson, Xiaoyan Sun, Simone Picelli, et al. 2016. “Single-Cell Transcriptome Profiling of Human Pancreatic Islets in Health and Type 2 Diabetes.” <em>Cell Metabolism</em> 24 (4): 593–607.</p>
</div>
<div id="ref-xin2016rna">
<p>Xin, Yurong, Jinrang Kim, Haruka Okamoto, Min Ni, Yi Wei, Christina Adler, Andrew J Murphy, George D Yancopoulos, Calvin Lin, and Jesper Gromada. 2016. “RNA Sequencing of Single Human Islet Cells Reveals Type 2 Diabetes Genes.” <em>Cell Metabolism</em> 24 (4): 608–15.</p>
</div>
</div>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
